beacon_node/http_api/src/lib.rs

```diff
@@ -2018,6 +2022,104 @@ pub fn serve<T: BeaconChainTypes>(
             })
         });
 
+    /*
+     * builder/states
+     */
+
+    let builder_states_path = eth_v1
+        .and(warp::path("builder"))
+        .and(warp::path("states"))
+        .and(chain_filter.clone());
+
+    // GET builder/states/{state_id}/expected_withdrawals
+    let get_expected_withdrawals = builder_states_path
+        .clone()
+        .and(warp::path::param::<StateId>())
+        .and(warp::path("expected_withdrawals"))
+        .and(warp::query::<api_types::ExpectedWithdrawalsQuery>())
+        .and(warp::path::end())
+        .and(warp::header::optional::<api_types::Accept>("accept"))
+        .and_then(
+            |chain: Arc<BeaconChain<T>>,
+             state_id: StateId,
+             query: api_types::ExpectedWithdrawalsQuery,
+             accept_header: Option<api_types::Accept>| {
+                blocking_response_task(move || {
+                    let (mut state, execution_optimistic, finalized) = state_id.state(&chain)?;
+                    let proposal_slot = query.proposal_slot.unwrap_or(state.slot() + 1);
+                    if proposal_slot < state.slot() {
+                        return Err(warp_utils::reject::custom_bad_request(
+                            "proposal slot must be greater than state slot".to_string(),
+                        ));
+                    }
+
+                    let fork = chain.spec.fork_name_at_slot::<T::EthSpec>(proposal_slot);
+                    if let ForkName::Base | ForkName::Altair | ForkName::Merge = fork {
+                        return Err(warp_utils::reject::custom_bad_request(
+                            "the specified state is not a capella state.".to_string(),
+                        ));
+                    }
+
+                    let look_ahead_limit = chain
+                        .spec
+                        .max_seed_lookahead
+                        .into_inner()
+                        .safe_mul(T::EthSpec::slots_per_epoch())
+                        .unwrap();
+                    if proposal_slot >= state.slot() + look_ahead_limit {
+                        return Err(warp_utils::reject::custom_bad_request(format!(
+                            "proposal slot cannot be >= the look ahead limit: {look_ahead_limit}"
+                        )));
+                    }
+
+                    let proposal_epoch = proposal_slot.epoch(T::EthSpec::slots_per_epoch());
+                    let (state_root, _, _) = state_id.root(&chain).unwrap();
```

Comment:
We should avoid `unwrap` here and return an error instead if state is not found.